storage-operator/pkg/controller/volume/volume_controller.go

```diff
@@ -88,85 +488,214 @@ type ReconcileVolume struct {
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileVolume) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Name", request.Name)
-	reqLogger.Info("Reconciling Volume")
+	reqLogger.Info("reconciling volume: START")
+	defer reqLogger.Info("reconciling volume: STOP")
 
-	// Fetch the Volume instance
-	instance := &storagev1alpha1.Volume{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Fetch the requested Volume object.
+	volume := &storagev1alpha1.Volume{}
+	err := r.client.Get(ctx, request.NamespacedName, volume)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			// TODO Remove the finalizer from the generated PV, if exists
-			return reconcile.Result{}, nil
+			// Volume not found:
+			// => all the finalizers have been removed & Volume has been deleted
+			// => there is nothing left to do
+			reqLogger.Info("volume already deleted: nothing to do")
+			return endReconciliation()
 		}
-		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		reqLogger.Error(err, "cannot read Volume: requeue")
+		return delayedRequeue(err)
 	}
-
-	result, err := r.salt.TestPing()
-	if err != nil {
-		reqLogger.Error(err, "test.ping failed")
-		return reconcile.Result{}, err
+	// This check should be done by an Admission Controller…
+	if err := volume.IsValid(); err != nil {
+		return r.setFailedVolumeStatus(
+			ctx, volume, storagev1alpha1.InternalError,
+			"invalid volume: %s", err.Error(),
+		)
 	}
-	for hostname := range result {
+	// Check if the volume is marked for deletion (i.e., deletion tsamp is set).
```

Comment:
`tsamp`?

Reply:
typo, meant `tstamp` as short for `timestamp`.

Done